provider/resources.go

```diff
@@ -490,6 +490,7 @@ var moduleMap = map[string]string{
 	"transfer":                        transferMod,
 	"verifiedaccess":                  verifiedaccessMod,
 	"verifiedpermissions":             verifiedpermissionsMod,
+	"vpc_endpoint_associations":       "VpcEndpointAssociations",
```

Comment:
I think "vpc_endpoint_associations" should be part of the "vpc" mod. We'll probably have to add an entry like this

https://github.com/pulumi/pulumi-aws/blob/a31fb4d445d76d0c8ceab67e259ac87ef2cb4e46/provider/resources.go#L4476-L4478

We've decided we want to organize the resource based on how terraform organizes them in their docs
https://github.com/pulumi/pulumi-aws/issues/5168

Reply:
It is part of the ec2 module upstream https://github.com/hashicorp/terraform-provider-aws/pull/41918/files probably most naturally belongs there. We have a little too much functionality in that module but still probably err on placing it there.

Reply:
I thought we had decided in #5168 that we should match how the Terraform docs organize things? For #5168 and v7 we need to decide what we base our organization on. For example, upstream doesn't even have a `vpc` folder.

Reply:
https://github.com/pulumi/pulumi-aws/pull/5161#discussion_r1942063652

Reply:
Hmm, sorry I have missed that. https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/vpc_endpoint.html is organized under VPC indeed. That sounds like a lot more intentional than where the source is located. Let me fix up.